makefile

```diff
@@ -1748,7 +1748,7 @@ $(build)/android.dep: $(luni-javas) $(dalvik-javas) $(libart-javas) \
 		| (cd $(build)/android-src && $(jar) x)
 #	(cd android && $(jar) c *)	| (cd $(build)/android-src && $(jar) x)
 	find $(build)/android-src -name '*.java' > $(build)/android.txt
-	$(javac) -Xmaxerrs 1000 -d $(build)/android @$(build)/android.txt
+	$(javac) -J-Xms256m -J-Xmx256m -Xmaxerrs 1000 -d $(build)/android @$(build)/android.txt
```

Comment:
A couple things:
* You're looking at a different value there than you're setting:  You're reading the Perm-gen size (which is just for stuff that will never be garbage collected, and should be small) - and you're setting the max total heap size (which doesn't have a fixed default; it'll default to 1/4 of your system memory).  What you want to grep for is 'MaxHeapSize', not 'MaxPermSize'.  On my workstation, I confirmed that it reports the default max size as ~8GB (I have 32GB memory).  256MB is substantially smaller than that.

Are you perhaps compiling on a VM with very little memory (512MB, or even less)?

If just bumping your system memory isn't an option, perhaps add a `$(JAVAC_OPTS)` instead of these switches, which will by default be empty, but on your system you can add those particular arguments you need by setting that environment variable (or make parameter, or whatever).